msprime/demography.py

```diff
@@ -0,0 +1,1091 @@
+            raise ValueError("Must specify one or more of growth_rate and initial_size")
+        if self.initial_size is not None and self.initial_size <= 0:
+            raise ValueError("Cannot have a population size <= 0")
+        self.population = -1 if self.population is None else self.population
+
+    def get_ll_representation(self):
+        ret = {
+            "type": "population_parameters_change",
+            "time": self.time,
+            "population": self.population,
+        }
+        if self.growth_rate is not None:
+            ret["growth_rate"] = self.growth_rate
+        if self.initial_size is not None:
+            ret["initial_size"] = self.initial_size
+        return ret
+
+    def __str__(self):
+        s = f"Population parameter change for {self.population}: "
+        if self.initial_size is not None:
+            s += f"initial_size -> {self.initial_size} "
+        if self.growth_rate is not None:
+            s += f"growth_rate -> {self.growth_rate} "
+        return s[:-1]
+
+
+@attr.s
+class MigrationRateChange(DemographicEvent):
+    """
+    Changes the rate of migration from one deme to another to a new value at a
+    specific time. Migration rates are specified in terms of the rate at which
+    lineages move from population ``source`` to ``dest`` during the progress of
+    the simulation. Note that ``source`` and ``dest`` are from the perspective
+    of the coalescent process; please see the :ref:`sec_api_simulation_model`
+    section for more details on the interpretation of this migration model.
+
+    By default, ``source=-1`` and ``dest=-1``, which results in all
+    non-diagonal elements of the migration matrix being changed to the new
+    rate. If ``source`` and ``dest`` are specified, they must refer to valid
+    population IDs.
+
+    :param float time: The time at which this event occurs in generations.
+    :param float rate: The new per-generation migration rate.
+    :param int source: The ID of the source population.
+    :param int dest: The ID of the destination population.
+    :param int source: The source population ID.
+    """
+
+    rate = attr.ib()
+    source = attr.ib(default=-1)
+    dest = attr.ib(default=-1)
+    # Deprecated.
+    # TODO add a formal deprecation notice
+    matrix_index = attr.ib(default=None, repr=False)
+
+    def __attrs_post_init__(self):
+        # If the deprecated form is used, it overwrites the values of source
+        # and dest
+        if self.matrix_index is not None:
+            self.source = self.matrix_index[0]
+            self.dest = self.matrix_index[1]
+
+    def get_ll_representation(self):
+        return {
+            "type": "migration_rate_change",
+            "time": self.time,
+            "migration_rate": self.rate,
+            "source": self.source,
+            "dest": self.dest,
+        }
+
+    def __str__(self):
+        if self.source == -1 is self.dest == -1:
```

Comment:
I know this isn't new code, just moved, but current line gives `SyntaxWarning` on py 3.8.
```suggestion
        if (self.source == -1) == (self.dest == -1)
```

Reply:
Eek! Not sure how that one snuck through.